src/main/java/org/openhab/binding/zigbee/converter/ZigBeeConverterColorColor.java

```diff
@@ -87,14 +156,105 @@ public void initializeConverter() {
+            clusterLevelControl.getCurrentLevel(0);
+        }
+
+        if (clusterOnOff != null) {
+            clusterOnOff.getOnOff(0);
+        }
+    }
+
+    private void changeOnOff(OnOffType onoff) throws InterruptedException, ExecutionException {
+        boolean on = onoff == OnOffType.ON;
+        PercentType brightness = on ? PercentType.HUNDRED : PercentType.ZERO;
+
+        if (clusterLevelControl != null) {
+            changeBrightness(brightness);
+            return;
+        }
+
+        if (clusterOnOff == null) {
+            logger.warn("{}: ignoring on/off command", endpoint.getIeeeAddress());
+            return;
+        }
+
+        HSBType oldHSB = currentHSB;
+        currentHSB = new HSBType(oldHSB.getHue(), oldHSB.getSaturation(), brightness);
+        lastBrightness = brightness;
+
+        if (on) {
+            clusterOnOff.onCommand().get();
+        } else {
+            clusterOnOff.offCommand().get();
+        }
+    }
+
+    private void changeBrightness(PercentType brightness) throws InterruptedException, ExecutionException {
+        if (clusterLevelControl == null) {
+            if (clusterOnOff != null) {
+                changeOnOff(brightness.intValue() == 0 ? OnOffType.OFF : OnOffType.ON);
+            } else {
+                logger.warn("{}: ignoring brightness command", endpoint.getIeeeAddress());
+            }
+            return;
+        }
+
+        HSBType oldHSB = currentHSB;
+        currentHSB = new HSBType(oldHSB.getHue(), oldHSB.getSaturation(), brightness);
+        lastBrightness = brightness;
+
+        int level = (int) (brightness.floatValue() * 254.0f / 100.0f + 0.5f);
+
+        if (clusterOnOff != null) {
+            clusterLevelControl.moveToLevelWithOnOffCommand(level, 10).get();
+        } else {
+            clusterLevelControl.moveToLevelCommand(level, 10).get();
+        }
+    }
+
+    private void changeColorHueSaturation(HSBType color) throws InterruptedException, ExecutionException {
+        HSBType oldHSB = currentHSB;
+        currentHSB = new HSBType(color.getHue(), color.getSaturation(), oldHSB.getBrightness());
+        int hue = (int) (color.getHue().floatValue() * 254.0f / 360.0f + 0.5f);
+        int saturation = (int) (color.getSaturation().floatValue() * 254.0f / 100.0f + 0.5f);
+
+        clusterColorControl.moveToHueAndSaturationCommand(hue, saturation, 10).get();
+    }
+
+    private void changeColorXY(HSBType color) throws InterruptedException, ExecutionException {
+        PercentType xy[] = ColorHelper.toXY(color);
+
+        HSBType oldHSB = currentHSB;
+        currentHSB = new HSBType(color.getHue(), color.getSaturation(), oldHSB.getBrightness());
+
+        logger.debug("{}: Change Color HSV. {}, {}, {}", endpoint.getIeeeAddress(), color.getHue(),
+                color.getSaturation(), oldHSB.getBrightness());
+        logger.debug("{}: Calculated XY. {}, {}", endpoint.getIeeeAddress(), xy[0], xy[1]);
```

Comment:
I'd suggest to make these a single log entry as really what you're wanting to see is the conversion and in the changeColorHueSaturation method, you aren't logging the "Change color HSV" line....